eclipse-collections/src/main/java/org/eclipse/collections/impl/map/ordered/mutable/OrderedMapAdapter.java

```diff
@@ -0,0 +1,809 @@
+import org.eclipse.collections.impl.block.procedure.primitive.CollectFloatProcedure;
+import org.eclipse.collections.impl.block.procedure.primitive.CollectIntProcedure;
+import org.eclipse.collections.impl.block.procedure.primitive.CollectLongProcedure;
+import org.eclipse.collections.impl.block.procedure.primitive.CollectShortProcedure;
+import org.eclipse.collections.impl.collection.mutable.CollectionAdapter;
+import org.eclipse.collections.impl.factory.Bags;
+import org.eclipse.collections.impl.list.fixed.ArrayAdapter;
+import org.eclipse.collections.impl.list.mutable.FastList;
+import org.eclipse.collections.impl.list.mutable.primitive.BooleanArrayList;
+import org.eclipse.collections.impl.list.mutable.primitive.ByteArrayList;
+import org.eclipse.collections.impl.list.mutable.primitive.CharArrayList;
+import org.eclipse.collections.impl.list.mutable.primitive.DoubleArrayList;
+import org.eclipse.collections.impl.list.mutable.primitive.FloatArrayList;
+import org.eclipse.collections.impl.list.mutable.primitive.IntArrayList;
+import org.eclipse.collections.impl.list.mutable.primitive.LongArrayList;
+import org.eclipse.collections.impl.list.mutable.primitive.ShortArrayList;
+import org.eclipse.collections.impl.map.AbstractMapIterable;
+import org.eclipse.collections.impl.map.mutable.UnifiedMap;
+import org.eclipse.collections.impl.map.mutable.primitive.ObjectDoubleHashMap;
+import org.eclipse.collections.impl.map.mutable.primitive.ObjectLongHashMap;
+import org.eclipse.collections.impl.multimap.list.FastListMultimap;
+import org.eclipse.collections.impl.partition.list.PartitionFastList;
+import org.eclipse.collections.impl.set.mutable.SetAdapter;
+import org.eclipse.collections.impl.tuple.AbstractImmutableEntry;
+import org.eclipse.collections.impl.utility.LazyIterate;
+import org.eclipse.collections.impl.utility.MapIterate;
+
+/**
+ * @since 9.2
+ */
+public class OrderedMapAdapter<K, V>
+        extends AbstractMapIterable<K, V>
+        implements MutableOrderedMap<K, V>, Serializable
+{
+    private static final long serialVersionUID = 1L;
+    private final Map<K, V> delegate;
+
+    protected OrderedMapAdapter(Map<K, V> delegate)
+    {
+        if (delegate == null)
+        {
+            throw new NullPointerException("OrderedMapAdapter may not wrap null");
+        }
+        this.delegate = delegate;
+    }
+
+    public static <K, V> MutableOrderedMap<K, V> adapt(Map<K, V> map)
+    {
+        return map instanceof MutableOrderedMap<?, ?> ? (MutableOrderedMap<K, V>) map : new OrderedMapAdapter<>(map);
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        return this.delegate.equals(o);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return this.delegate.hashCode();
+    }
+
+    @Override
+    public MutableOrderedMap<K, V> newEmpty()
+    {
+        return OrderedMapAdapter.adapt(new LinkedHashMap<>());
+    }
+
```

Comment:
This is an adapter, so there are no public constructors. This comment is on newEmpty(), which is awkward on adapters.

Reply:
I just looked and MapAdapter does have newEmpty(capacity), but it doesn't come from any interface. Thoughts?

Reply:
For minor release let us add it on OrderedMapAdapter, and then eventually move it up to MutableMapIterable in 10.0.0. How does that sound?

Reply:
Since this is an adapter, the usual use-case will probably be wrapping existing LinkedHashMaps, which already have their own capacity. Eventually we'd want to implement our own MutableOrderedMap which isn't an adapter, and then transformation methods would return this type. Just like how ListAdapter.select() return a FastList, not a ListAdapter. That type would have a constructor with a capacity.

The other odd thing about putting it on the interface is that several collections cannot support a capacity, specifically the ones based on trees. That said, it's reasonable to just ignore the capacity.